docs/tutorials/migrate/migrate_from_mbed_os_2_to_5.md

```diff
@@ -0,0 +1,308 @@
+
+First, navigate to the Hello World repository for the component.
+
+<span class="images">![](https://os.mbed.com/media/uploads/jplunkett/1-helloworld.png)</span>
+
+Next, view the files in the Hello World repository. The presence of an  `mbed.bld` or `mbed-rtos.lib` file signifies that this program uses an older version of Mbed OS (older than Mbed OS 5).
+
+<span class="images">![](https://os.mbed.com/media/uploads/jplunkett/2-oldmbed.png)</span>
+
+A program that uses and has been tested with Mbed OS 5 or later has an `mbed-os.lib` file.
+
+<span class="images">![](https://os.mbed.com/media/uploads/jplunkett/3-newmbed.png)</span>
+
+Some repositories may have both an `mbed.bld` file and a `mbed-rtos.lib` file present. Mbed-RTOS is the precursor to Mbed OS 5, which combines the older Mbed OS 2 library with Mbed-RTOS. So, Mbed OS 5 can replace BOTH `mbed.bld` and `mbed-rtos.lib`.
+
+### Migrating to Mbed OS 5
+
+There are two possible outcomes when trying to migrate to Mbed OS 5:
+
+1. Replacing the old Mbed library with Mbed OS "just works," as the APIs in the library have not changed. In this instance, you're finished.
+2. Replacing the old Mbed library with Mbed OS produces some compilation errors. For one of two reasons:
+  - The API calls in the **library** are out of date, and you need to migrate to Mbed OS API syntax.
+  - The Hello World **application** uses target specific code, and you need to be migrate it to use code for the target you are compiling for.
+
+The general outline for updating to Mbed OS 5 is:
+
+```
+mbed import [URL of Hello World]
+cd [Project Name]
+mbed remove mbed
+mbed remove mbed-rtos
+mbed add mbed-os
+```
+
+To determine the success of migration, run:
+
+`mbed compile -m [platform] -t [toolchain]`
+
+### Example component #1 - migrating just works
+
+- Component: [Grove - Buzzer](https://os.mbed.com/components/Grove-Buzzer/).
+- Hello World repo: [Seeed_Grove_Buzzer](https://os.mbed.com/teams/Seeed/code/Seeed_Grove_Buzzer/).
+- Platform: [u-blox EVK-ODIN-W2](https://os.mbed.com/platforms/ublox-EVK-ODIN-W2/).
+- Toolchain: [GCC ARM](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads).
+
+In the command-line, run:
+
+```
+mbed import https://os.mbed.com/teams/Seeed/code/Seeed_Grove_Buzzer/
+cd Seeed_Grove_Buzzer
+mbed remove mbed
+mbed remove mbed-rtos
+mbed add mbed-os
+mbed compile -m ublox_evk_odin_w2 -t gcc_arm
+```
+
+It successfully compiles, so no changes are necessary to the Grove - Buzzer library or Hello World program.
+
+### Example component #2 - application fails to compile
+
+- Component: [SRF08-Ultrasonic-Range-Finder](https://os.mbed.com/components/SRF08-Ultrasonic-Range-Finder/).
+- Hello World repo: [SRF08HelloWorld](https://os.mbed.com/users/melse/code/SRF08HelloWorld/).
+- Platform: [FRDM-K64F](https://os.mbed.com/platforms/FRDM-K64F/).
```

Comment:
I think it should be more clear that the platform/toolchain listed are what are using to compile and not the only supported target/toolchain for this example.

It is obvious later in the compile command, but I think it should be clear when they are first listed.

Reply:
Would something like this work?

```
Repositories and configuration information used for this example:

- Component: repo
- ... etc.
```

Reply:
I would split it into:

Repositories
- repo1
- repo2

Compile conditions/settings/something:
- GCC 
- K64F